### PR TITLE
[E-nodes < T500] Implement nodes.relationships.exist

### PR DESCRIPTION
**Pull request by mpintaric55334**
### Description

Please briefly explain the changes you made here.

### Pull request type

- [ ] Bugfix
- [x] Algorithm/Module
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

### Related issues

Delete if this PR doesn't resolve any issues. Link the issue if it does.

######################################

#### Module/Algorithm
- [ ] Core algorithm/module implementation
- [ ] [Query module](https://memgraph.com/docs/memgraph/reference-guide/query-modules) implementation
- [ ] Unit tests
- [ ] End-to-end tests
- [ ] Code documentation
- [ ] README short description
- [ ] Documentation on [memgraph/docs](https://github.com/memgraph/docs)
- [ ] Update GQLALchemy signatures in [query builder](https://github.com/memgraph/gqlalchemy/blob/main/gqlalchemy/graph_algorithms/query_builder.py  ) using [query module signature generator](https://github.com/memgraph/gqlalchemy/blob/main/scripts/query_module_signature_generator.py)

######################################

### CODE REVIEW

File: e2e/nodes_test/test_relationshipsExist1/test.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+query: >
+    MATCH (p1:Person)
+    WHERE p1.name IN ["Michael", "Person30", "Person60"]
+    WITH collect(p1) AS people
+    CALL nodes.relationships_exist(people, ["KNOWS>","FOLLOWS","<KNOWS"]) YIELD result RETURN result["Relationships_exist_status"];
```
Comment (Josipmrden) [inline e2e/nodes_test/test_relationshipsExist1/test.yml:5]:
```suggestion
    CALL nodes.relationships_exist(people, ["KNOWS>","FOLLOWS","<KNOWS"]) YIELD result RETURN result["Relationships_exist_status"] AS status;
```

File: e2e/nodes_test/test_relationshipsExist1/test.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+output:
+  - result["Relationships_exist_status"]: {"<KNOWS": false, "FOLLOWS": true,"KNOWS>": true}
+  - result["Relationships_exist_status"]: {"<KNOWS": true, "FOLLOWS": true,"KNOWS>": false}
+  - result["Relationships_exist_status"]: {"<KNOWS": true, "FOLLOWS": false,"KNOWS>": false}
```
Comment (Josipmrden) [inline e2e/nodes_test/test_relationshipsExist1/test.yml:9]:
```suggestion
  - status: {"<KNOWS": false, "FOLLOWS": true,"KNOWS>": true}
  - status: {"<KNOWS": true, "FOLLOWS": true,"KNOWS>": false}
  - status: {"<KNOWS": true, "FOLLOWS": false,"KNOWS>": false}
```

File: e2e/nodes_test/test_relationshipsExist2/test.yml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+query: >
+    MATCH (d:Dog), (h:Human)
+    CALL nodes.relationships_exist([d,h], ["Loves>","Loves"]) YIELD result RETURN result["Relationships_exist_status"];
+output:
+  - result["Relationships_exist_status"]: {"Loves": true,"Loves>": true}
+  - result["Relationships_exist_status"]: {"Loves": true,"Loves>": false}
```
Comment (Josipmrden) [inline e2e/nodes_test/test_relationshipsExist2/test.yml:6]: same like previous comment

File: e2e/nodes_test/test_relationshipsExist3/test.yml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+query: >
+    MATCH (h:Human)
+    CALL nodes.relationships_exist([h], ["Loves>","chases>"]) YIELD result RETURN result["Relationships_exist_status"];
+output:
+  - result["Relationships_exist_status"]: {"Loves>": false,"chases>": true}
```
Comment (Josipmrden) [inline e2e/nodes_test/test_relationshipsExist3/test.yml:5]: same like previous comment

File: cpp/nodes_module/algorithm/nodes.hpp [hunk 5]
```diff
@@ -1,5 +1,15 @@
...
-namespace Nodes {}  // namespace Create
\ No newline at end of file
+namespace Nodes {
+
+/*relationships_exist constants*/
+constexpr std::string_view kProcedureRelationshipsExist= "relationships_exist";
+constexpr std::string_view kReturnRelationshipsExist= "result";
+constexpr std::string_view kArgumentNodesRelationshipsExist= "nodes";
+constexpr std::string_view kArgumentRelationshipsRelationshipsExist= "relationships";
+void RelationshipsExist(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory);
```
Comment (Josipmrden) [inline cpp/nodes_module/algorithm/nodes.hpp:13]: space between `constexpr` and functions

File: cpp/nodes_module/algorithm/nodes.cpp [hunk 6]
```diff
@@ -1 +1,59 @@
...
+  }else if (rel_type[0] == '<') {
+    direction = rel_type[0];
+    rel_type.erase(0,1);
+  }
+  for (auto rel : node.OutRelationships()) {
+    if (std::string(rel.Type()) == rel_type && direction != '<') {
+      return true;
+    }
+  }
+  for (auto rel : node.InRelationships()) {
```
Comment (Josipmrden) [inline cpp/nodes_module/algorithm/nodes.cpp:19]: const auto &rel

File: cpp/nodes_module/algorithm/nodes.cpp [hunk 7]
```diff
@@ -1 +1,59 @@
...
+  if (rel_type[0] == '<' && rel_type[rel_type.size() - 1] == '>') {
+    throw mgp::ValueException("Invalid relationship specification!");
+  }else if (rel_type[rel_type.size() - 1] == '>') {
+    direction = rel_type[rel_type.size() - 1];
+    rel_type.pop_back();
+  }else if (rel_type[0] == '<') {
+    direction = rel_type[0];
+    rel_type.erase(0,1);
+  }
+  for (auto rel : node.OutRelationships()) {
```
Comment (Josipmrden) [inline cpp/nodes_module/algorithm/nodes.cpp:14]: const auto &rel

File: cpp/nodes_module/algorithm/nodes.cpp [hunk 8]
```diff
@@ -1 +1,59 @@
-#include "nodes.hpp"
\ No newline at end of file
+#include "nodes.hpp"
+
+bool Nodes::RelationshipExist(const mgp::Node &node, std::string &rel_type) {
+  char direction{' '};
+  if (rel_type[0] == '<' && rel_type[rel_type.size() - 1] == '>') {
+    throw mgp::ValueException("Invalid relationship specification!");
+  }else if (rel_type[rel_type.size() - 1] == '>') {
```
Comment (Josipmrden) [inline cpp/nodes_module/algorithm/nodes.cpp:7]: space between `}` and `else if`

File: cpp/nodes_module/algorithm/nodes.cpp (hunk 7 above)
Comment (antoniofilipovic) [reply cpp/nodes_module/algorithm/nodes.cpp:14]: I think we don't need ref as Relationship object is returned as Value, there isn't reference to something that exists.

File: cpp/nodes_module/algorithm/nodes.cpp (hunk 6 above)
Comment (antoniofilipovic) [reply cpp/nodes_module/algorithm/nodes.cpp:19]: I think we don't need ref as Relationship object is returned as Value, there isn't reference to something that exists.

**Review by antoniofilipovic — CHANGES_REQUESTED** [review]
I think you have few changes from @Josipmrden

**Review by antepusic — CHANGES_REQUESTED** [review]
Left a few comments :incoming_envelope:

File: cpp/nodes_module/algorithm/nodes.cpp [hunk 9]
```diff
@@ -1 +1,64 @@
...
+    const mgp::Graph graph = mgp::Graph(memgraph_graph);
+    const mgp::List nodes = arguments[0].ValueList();
+    const mgp::List relationships = arguments[1].ValueList();
+    if (nodes.Size() == 0 || relationships.Size() == 0) {
+      throw mgp::ValueException("Input lists must not be empty!");
+    }
+
+    for (auto element: nodes) {
+      if(!element.IsNode() && !element.IsInt()){
+        throw mgp::ValueException("Input arguments must be nodes or their ID's");
```
Comment (antepusic) [inline cpp/nodes_module/algorithm/nodes.cpp:41]:
```suggestion
        throw mgp::ValueException("Input arguments must be nodes or their IDs");
```

**Review by antepusic — APPROVED** [review]
Just resolve the conflicts and you’re good to go :man_technologist:

**Review by antepusic — COMMENTED** [review]
After resolving the conflicts, please make sure that you run checks and they pass

**Review by antepusic — CHANGES_REQUESTED** [review]
NB: This is not about code (to me it’s approvable), but the checks weren’t run because of the conflicts with `main`. Make sure they pass and it’s an approve from me :smile:

**Review by antoniofilipovic — COMMENTED** [review]
I think file names for tests can be a bit different: `test_relationshipsExist1` -> `test_relationship_exist_1`. Does this make sense consistency-wise?

**Review by antoniofilipovic — APPROVED** [review]

**Review by antepusic — APPROVED** [review]
Approving :smile: 

NB: The reason for the checks not passing is because #324 has not been merged into this PR’s branch.

**Pull request merged**